runtime/oti/j9nonbuilder.h

```diff
@@ -5780,8 +5780,10 @@ typedef struct J9JavaVM {
 	struct J9HashTable* ensureHashedClasses;
 #if JAVA_SPEC_VERSION >= 19
 	U_64 nextTID;
-	j9object_t *liveVirtualThreadList;
-	omrthread_monitor_t liveVirtualThreadListMutex;
+	j9object_t *virtualThreadList;
+	omrthread_monitor_t virtualThreadInspectorMutex;
+	omrthread_monitor_t virtualThreadListMutex;
+	volatile BOOLEAN inspectVirtualThreadList;
```

Comment:
Should this be a counter?

Reply:
Currently, for functional correctness, only one thread is allowed to resumeAll or suspendAll vthreads at any time. So, a boolean should suffice for this purpose. Counter is not required since multiple threads are not allowed to inspect the list simultaneously.

Reply:
That's my point - this seems unnecessarily restrictive. There's no one else who wants to examine the list? Looks like this is also protecting modification of the list (as it must).

Reply:
> no one else who wants to examine the list?

[Resume|Suspend]AllVirtualThreads can be invoked from different threads simultaneously. It is to protect vthreads from being resumed and suspended simultaneously, which will lead to undefined behaviour.

Reply:
The use of mulitple mutexxes is also confusing? Why can't one be used to protect both the list and the flag/counter?

Reply:
Will attempt to reimplement with one mutex; changing to draft.